sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -206,4 +216,12 @@ class DatasetAggregatorSuite extends QueryTest with SharedSQLContext {
       ds.groupByKey(_.b).agg(ComplexBufferAgg.toColumn),
       ("one", 1), ("two", 1))
   }
+
+  test("SPARK-12555 - data is not corrupted after input data is reordered") {
```

Comment:
yea, but we need to make the test name describe the problem more clearly, how about `SPARK-12555 - result should not be corrupted after input columns are reordered`?